DSCResources/MSFT_xVMHyperV/MSFT_xVMHyperV.psm1

```diff
@@ -495,6 +528,71 @@ function Change-VMProperty
     }
 }
 
+function Change-VMSecureBoot
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [String]$Name,
+
+        [Boolean]$SecureBoot,
+
+        [Boolean]$RestartIfNeeded
+    )
+
+    $vmObj = Get-VM -Name $Name
+    $originalState = $vmObj.state
+    if($originalState -ne "Off" -and $RestartIfNeeded)
+    { 
+        Set-VMState -Name $Name -State Off
+        if ($SecureBoot)
+        {
+            Set-VMFirmware -VMName $Name -EnableSecureBoot On
+        }
+        else {
+            Set-VMFirmware -VMName $Name -EnableSecureBoot Off
+        }
+
+        # Can not move a off VM to paused, but only to running state
+        if($originalState -eq "Running")
+        {
+            Set-VMState -Name $Name -State Running -WaitForIP $true
+        }
+
+        Write-Verbose -Message "VM $Name now has correct properties."
+
+        # Cannot make a paused VM to go back to Paused state after turning Off
+        if($originalState -eq "Paused")
+        {
+            Write-Warning -Message "VM $Name state will be OFF and not Paused"
+        }
+    }
+    elseif($originalState -eq "Off")
+    {
+        if ($SecureBoot)
+        {
+            Set-VMFirmware -VMName $Name -EnableSecureBoot On
+        }
+        else {
+            Set-VMFirmware -VMName $Name -EnableSecureBoot Off
+        }
+    }
+    else
+    {
+        Write-Error -Message "Can not change properties for VM $Name in $($vmObj.State) state unless RestartIfNeeded is set to true"
+    }
+}
+
+function Get-VMSecureBoot
```

Comment:
This function should be called **Test-VMSecureBoot** since it is returning **$true** or **$false**